index.js

```diff
@@ -10,41 +10,79 @@ const path = require("path");
 const loaderUtils = require("loader-utils");
 const SourceNode = require("source-map").SourceNode;
 const SourceMapConsumer = require("source-map").SourceMapConsumer;
-const FOOTER = "/*** REQUIRES FROM autoload-plugins-loader ***/\n";
+const matchAll = require('match-all');
 
-module.exports = function(content, sourceMap) {
-    const resourcePath = this.resourcePath;
+const FOOTER = "/*** REQUIRES FROM autoload-modules-loader ***/\n";
+
+const REQUIRE_RE = /\brequire\s*\(\s*[\'"]([^\'"]*)[\'"]\s*\)/g;
 
+module.exports = function(content, sourceMap) {
     if (this.cacheable) {
         this.cacheable();
     }
 
-    const query = loaderUtils.getOptions(this) || {};
-    const requires = [];
-    const keys = Object.keys(query);
-
-    if (keys.length == 1 && typeof query[keys[0]] == "boolean") {
-        const mod = keys[0];
-        const modPath = path.resolve(__dirname, mod);
-        requires.push(`require("${modPath}");   // ${mod}`);
-    } else {
-        keys.forEach(function(name) {
-            const mod = name;
-            const modPath = path.resolve(__dirname, mod);
-            requires.push(`require("${modPath}");   // ${mod}`);
+    // promisify-node doesn't know how to deal with methods so we have to wrap
+    // this.resolve() ourselves.
+    const resolve = (context, request) =>
+        new Promise((resolve, reject) => {
+            this.resolve(context, request, (err, result) => {
```

Comment:
You don't necessarily have to do it, but `util.promisify` might be worth investigating in the future! https://nodejs.org/api/util.html#util_util_promisify_original

Reply:
I'll check it out, thanks!